File: lc_go/lcGo/dbpointer/countPairs.go
package main

import "slices"

func countPairs(nums []int, target int) (ans int) {
	slices.Sort(nums)
	i, j := 0, len(nums)-1
	for i < j {
		if nums[i]+nums[j] < target {
			ans += j - i
			i++
		} else {
			j--
		}
	}
	return
}

func main() {

}

File: lc_py/lcPy/double_pointer/countPairs.py
from typing import List


class Solution:
    def countPairs(self, nums: List[int], target: int) -> int:
        nums.sort()
        ans, i, j = 0, 0, len(nums) - 1

        while i < j:
            if nums[i] + nums[j] < target:
                ans += (j - i)
                i += 1
            else:
                j -= 1
        return ans